### ISSUE TITLE
Oncoprint Uncaught SyntaxError and wrong parsing of gene name EXPH5

### ISSUE THREAD

**Issue by pieterlukasse** [cBioPortal/cbioportal#1290]
When executing [this query](http://www.cbioportal.org/beta/index.do?cancer_study_list=brca_tcga&cancer_study_id=brca_tcga&genetic_profile_ids_PROFILE_MUTATION_EXTENDED=brca_tcga_mutations&genetic_profile_ids_PROFILE_COPY_NUMBER_ALTERATION=brca_tcga_gistic&genetic_profile_ids_PROFILE_MRNA_EXPRESSION=brca_tcga_rna_seq_v2_mrna_median_Zscores&Z_SCORE_THRESHOLD=2.0&RPPA_SCORE_THRESHOLD=2.0&data_priority=0&case_set_id=brca_tcga_3way_complete&case_ids=&patient_case_select=sample&gene_set_choice=user-defined-list&gene_list=AAVS1%250D%250AACAN%250D%250AAMY1A%250D%250AAPOA4%250D%250ABRD1%250D%250AC4A%250D%250ACALM2%250D%250ACEL%250D%250ACHRNA7%250D%250ACKMT1B%250D%250ADNAH17%250D%250AEXPH5%250D%250AGABARAPL3%250D%250AGAGE2C%250D%250AGAS8-AS1%250D%250AGPR42%250D%250AH3F3A%250D%250AHELZ%250D%250AHIST1H2AE%250D%250AHIST1H2AI%250D%250AHIST1H2AK%250D%250AHIST1H2AL%250D%250AHIST1H2AM%250D%250AHIST1H2BE%250D%250AHIST1H2BF%250D%250AHIST1H2BG%250D%250AHIST1H3A%250D%250AHIST1H3B%250D%250AHIST1H3C%250D%250AHIST1H3D%250D%250AHIST1H3G%250D%250AHIST1H3I%250D%250AHIST1H3J%250D%250AHIST1H4B%250D%250AHIST1H4D%250D%250AHIST1H4H%250D%250AHIST1H4J%250D%250AHIST1H4L%250D%250AIFNA1%250D%250AIGHA1%250D%250AIGHA2%250D%250AIGHD%250D%250AIGHE%250D%250AIGHG2%250D%250AIGHG3%250D%250AIGHM%250D%250AIGHV3-38%250D%250AIGHV3-43%250D%250AIGHV3-48%250D%250AIGHV3-49%250D%250AIGHV3-53%250D%250AIGHV3-66%250D%250AIGHV3-72%250D%250AIGHV3-73%250D%250AIGHV3-74%250D%250AIGHV4-28%250D%250AIGHV4-31%250D%250AIGHV4-59%250D%250AIGHV5-51%250D%250AIGHV6-1%250D%250AIGHV7-81%250D%250AIGKC%250D%250AIGKV1-5%250D%250AIGLC2%250D%250AIGLC3%250D%250AKIAA0087%250D%250AKRT4%250D%250ALILRB2%250D%250ALINC01558%250D%250ALPA%250D%250AMUC3A%250D%250AMUC5AC%250D%250ANBPF14%250D%250AOPN1LW%250D%250AOPN1MW%250D%250AOR10D3%250D%250AOR10G2%250D%250AOR4E2%250D%250API4KA%250D%250APPFIA4%250D%250APRB3%250D%250APRH1%250D%250APTPRH%250D%250APZP%250D%250ARORB%250D%250ASEMA4B%250D%250ASLC5A3%250D%250ASTON1%250D%250ATMSB15A%250D%250ATRAV6%250D%250ATRBV20OR9-2%250D%250ATRGC1%250D%250ATRGJP%250D%250ATRGV1%250D%250ATRGV5%250D%250AUBBP4%250D%250AUGT2A1%250D%250AZNF66%250D%250AZRSR1%250D%250AZSCAN12&clinical_param_selection=null&tab_index=tab_visualize&Action=Submit&show_samples=false&), the oncoprint page stops working. In the console you can find the following error: 

![image](https://cloud.githubusercontent.com/assets/2900303/15575503/8b8587ea-2353-11e6-99ab-8544d0fda050.png)

**Comment by pieterlukasse** [comment]
I thought these were the steps to reproduce...but then I found another strange issue:

- go to homepage
- select TCGA Breast cancer study (provisional) - brca_tcga
- fill in these genes in the query:
```
AAVS1
ACAN
AMY1A
APOA4
BRD1
C4A
CALM2
CEL
CHRNA7
CKMT1B
DNAH17
EXPH5
GABARAPL3
GAGE2C
GAS8-AS1
GPR42
H3F3A
HELZ
HIST1H2AE
HIST1H2AI
HIST1H2AK
HIST1H2AL
HIST1H2AM
HIST1H2BE
HIST1H2BF
HIST1H2BG
HIST1H3A
HIST1H3B
HIST1H3C
HIST1H3D
HIST1H3G
HIST1H3I
HIST1H3J
HIST1H4B
HIST1H4D
HIST1H4H
HIST1H4J
HIST1H4L
IFNA1
IGHA1
IGHA2
IGHD
IGHE
IGHG2
IGHG3
IGHM
IGHV3-38
IGHV3-43
IGHV3-48
IGHV3-49
IGHV3-53
IGHV3-66
IGHV3-72
IGHV3-73
IGHV3-74
IGHV4-28
IGHV4-31
IGHV4-59
IGHV5-51
IGHV6-1
IGHV7-81
IGKC
IGKV1-5
IGLC2
IGLC3
KIAA0087
KRT4
LILRB2
LINC01558
LPA
MUC3A
MUC5AC
NBPF14
OPN1LW
OPN1MW
OR10D3
OR10G2
OR4E2
PI4KA
PPFIA4
PRB3
PRH1
PTPRH
PZP
RORB
SEMA4B
SLC5A3
STON1
TMSB15A
TRAV6
TRBV20OR9-2
TRGC1
TRGJP
TRGV1
TRGV5
UBBP4
UGT2A1
ZNF66
ZRSR1
ZSCAN12
```
- now when submitting, you get a red label on top of the query box saying 
`Error: Expression specified in the list of genes, but not selected in the Genetic Profile Checkboxes.`

**Comment by pieterlukasse** [comment]
this last issue seems to be related to gene `EXPH5`

**Comment by adamabeshouse** [comment]
I figured out the EXP issue. I see the error when I click on your URL, but I wasn't able to reproduce the other one with your query through the UI. How did you get there?

**Comment by pieterlukasse** [comment]
@adamabeshouse : I could also not reproduce it any more. I got there by pasting the genes in another server first (our test server cbioportal-test.thehyve.net/cbioportal), and then adjusting the first part of the url to point to your server. So it could be related to some difference in versions. Anyway, it is still best if any exception is caught and displayed to the user instead of letting the UI hang.

**Comment by adamabeshouse** [comment]
@pieterlukasse Good idea.

**Comment by pieterlukasse** [comment]
both issues now seem to be solved (I tested in `rc` after merge of #1416). Thanks!
